source/beamline/PGM/PGMBPMControl.cpp

```diff
@@ -0,0 +1,41 @@
+#include "beamline/PGM/PGMBPMControl.h"
+
+PGMBPMControl::PGMBPMControl(const QString &name, const QString &pvName, int avgValue, int variance, QObject *parent, const QString description)
+	: AMReadOnlyPVControl(name, pvName, parent, description)
+
+{
+    averageValue_ = avgValue;
+    variance_ = variance;
+	validityState_ = true;
+
+	// Get updated value from AMReadOnlyPVControl.
+	connect(this, SIGNAL(valueChanged(double)), this, SLOT(onPVValueChanged(double)));
+
+}
+
+QString PGMBPMControl::valueStr() const
```

Comment:
This would be a helper function for the receiver's convenience, if they want the value + unit format. If they want some thing else, they can listen to the valueChanged() signal and do it themselves.